pymongo/pool.py

```diff
@@ -237,6 +237,12 @@ def _raise_connection_failure(address, error, msg_prefix=None):
         msg = msg_prefix + msg
     if isinstance(error, socket.timeout):
         raise NetworkTimeout(msg) from error
+    elif isinstance(error, _SSLError) and 'timed out' in str(error):
+        # Eventlet does not distinguish TLS network timeouts from other
```

Comment:
I am wondering since this is only occurring when `eventlet` is installed should we check whether `eventlet` is installed (I don't know if its possible - I feel like it should be) inside this `elif`?

Reply:
No we can't check if eventlet is installed. Merely importing eventlet can have effects like we safe in the dnspython 2.0 vs eventlet bug.

Reply:
I see.
Using `__import__` will actually eliminate any side effects as it doesn't mess with the current global/local namespaces (unless instructed to do so) but I take your point that checking `eventlet` in any form would present a risk with minimal reward.

Reply:
I don't think `__import__` can prevent the types of issues we saw in https://github.com/mongodb/mongo-python-driver/pull/484#issuecomment-691224196